### PR TITLE
Tbolt/4506 Updates to proposed budget and executive summary 

### LINKED ISSUE #4506: [Feature] Updates to Proposed Budget and Executive Summary
### Description and related issues
Base on design ticket #3380

rows. It will help users follow the content across a row better. It is however not beneficial for smaller datasets because there is less of an issue with scanning and alternating rows, may give off the impression that we are highlighting some information and not others. From a dev perspective, we need a way to conditionally add alt rows if rows are greater
than 4 Even numbers would have the #F1F1F1 background color.

### Mockups or link to Figma
https://www.figma.com/file/hJpKHKU6fz5J0Z7fisSwa2/eAPD-MMIS-2022?node-id=2779:17295&t=XPkclhsJgd2SnnqZ-0

### Acceptance criteria

- [x] change the help text below the Summary Budget Table
> The following tables summarize the calculated Medicaid Total Computable (Total Activity Cost minus any Other Funding amounts) for key state personnel, state personnel, private contractor, and other state expenses for all activities by FFY. The tables provide subtotals for each of these categories by program area with the calculated Federal share by FFP rate and the associated state share. The final table in this section provides the total Medicaid cost for all activities in this APD 
by FFY.

- [x] update tables to display rows with alternating white and gray (#F1F1F1) rows if the table has more than 4 rows (count includes subtotal/total row). Even numbered rows should be gray

- #### Activity Total Costs changes
  - [x] Grand totals or total medicaid computable would have a dark yellow (#FAD980)
  - [x] Subtotals will be a lighter yellow (#FDF9EB)

- #### Budget and FFP table changes
  - [x] Grand totals or total medicaid computable would have a dark yellow (#FAD980)
  - [x] Subtotals will be a lighter yellow (#FDF9EB)

- #### Executive Summary Table changes
  - [x] Grand totals or total medicaid computable would have a dark yellow (#FAD980)
  - [x] Subtotals will be a lighter yellow (#FDF9EB)

- #### Combined Activity Costs changes
  - [x] Grand totals or total medicaid computable would have a dark yellow (#FAD980)
  - [x] Subtotals will be a lighter yellow (#FDF9EB)

- #### Summary Budget Table changes
  - [x] Grand totals or total medicaid computable would have a dark yellow (#FAD980)
  - [x] Subtotals will be a lighter yellow (#FDF9EB)

- #### State and Contractor Cost Breakdown changes
  - [x] Grand totals or total medicaid computable would have a dark yellow (#FAD980)
  - [x] Subtotals will be a lighter yellow (#FDF9EB)

- #### Make tables more consistent
if a table has a single header it should be medium gray (#D9D9D9), if a table has two headers the first header should be dark gray (#404040) and the second header should be medium gray (#D9D9D9)
  - [x] change Activity Schedule Summary table so that the header is the medium gray (#D9D9D9) with black text
CleanShot 2023-02-01 at 16 28 35
  - [x] change Program Budget Tables so that the first header is (#404040) and the second header stays (#D9D9D9)
- [x] Add lighter yellow to total computable column (#FDF9EB)
CleanShot 2023-02-01 at 16 34 28

### Testing criteria

| Given | When | Then | Covered |
| ----- | ---- | ---- | ------- |
|a HITECH or MMIS APD|On the Proposed Budget page, Executive Summary page, Budget and FFP page, and the Activity Schedule Summary page | All changes to tables should only affect the Activity total cost table, Combined Activity Cost table,  State and Contractor Cost breakdown table, Summary Budget Table, Budget and FFP table, Executive Summary Table, Schedule Summary by Activity, and Milestone Tables by Activity |(Add checkmark here when covered)|
|a HITECH or MMIS APD|On the Proposed Budget page, Executive Summary page, Budget and FFP page, and the Activity Schedule Summary page|If a table has more than 4 rows, each row should alternate in color. Even numbered rows should be gray (#F1F1F1) ||
|a HITECH or MMIS APD|Tables on the Proposed Budget page and Budget and FFP page| Subtotal rows should have a light yellow background color and Grand total rows should have a dark yellow background color| |
|a HITECH or MMIS APD|Tables on the Executive Summary page| Total row should be light yellow. This is an exception to the totals/subtotals color scheme. (Verified w/ Akua)| |
|a HITECH or MMIS APD|On the Proposed Budget page| Help text for the Summary Budget table section should be updated| |
|a HITECH or MMIS APD| Tables on the Activity Schedule Summary; Combined Activity Costs and State and Contractor Cost Breakdown tables on the Proposed budget | The table header background color should change from black to gray to match the other tables | |
|a HITECH or MMIS APD| Summary Budget Tables on the Proposed Budget | Federal total should come first before State total | |
|a HITECH or MMIS APD| Milestone Tables by Activity on the Activity Schedule Summary page | Table header should be changed to the activity name. Subheaders should be Milestone List and Target completion date | |

### PR DESCRIPTION
**Pull request by tbolt**
Resolves #4506 

### Description
Updates to table designs

[Chromatic approval ](https://www.chromatic.com/pullrequest?appId=61d5b948cf6f17003a12bf77&number=4583)

### Significant changes or possible side effects
* Styling changes for many of the table classes. Possible side-effects: Updating a CSS class has a cascading effect and updates an unintended component/table

### Automated test cases written
No new tests needed for these design changes. Tests were updated where needed.

### Steps to manually verify this change
1. Login to view APDs
2. View an APD with MMIS type
3. Verify Changes are according to ticket + Figma
4. Repeat with an HITECH APD type
5. Since we changed the order of the tables - swapping the federal and state share columns - test that state share data and federal share data is displaying in their correct columns

### This pull request is ready to code review when

- [x] Automated tests are updated (and all tests are passing)
- [x] New automated test cases are documented above
- [ ] Pull request has been labeled, if applicable with feature, content, bug,
      tests, refactor
- [ ] Associated OpenAPI documentation has been updated
- [ ] The experience passes a basic manual accessibility audit (keyboard nav,
      screenreader, text scaling) OR an exemption is documented

### This pull request is ready to test when

- [ ] Code has been reviewed by someone other than the original author

### This pull request is ready to review when the QA has

- [ ] Verified the functionality related to the change
- [ ] Verified that the change works with Narrator on Windows
- [ ] Verified that the change works with VoiceOver on Mac
- [ ] Verified all updated pages with the WAVE tool
- [ ] Verified tab and keyboard navigation functionality

### This pull request can be merged when

- [ ] Design has approved the experience
- [ ] Product has approved the experience

### CODE REVIEW

**Review by thetif — APPROVED** [review]
looks good

File: web/src/pages/apd/proposed-budget/BudgetSummary.js [hunk 1]
```diff
@@ -93,127 +107,146 @@ HeaderRow.propTypes = {
...
+              State Total
+            </th>
+            <th scope="col" className="ds-u-text-align--right">
+              Medicaid Total Computable
+            </th>
+          </tr>
+        </thead>
+        <tbody>
+          {orderedTotals.map((rowData, index) => {
+            console.log('rowData', rowData);
```
Comment (thetif) [inline web/src/pages/apd/proposed-budget/BudgetSummary.js:219]: console log

**Review by thetif — APPROVED** [review]
looks great, just one console log to remove

File: web/src/pages/apd/proposed-budget/BudgetSummary.js (hunk 1 above)
Comment (tbolt) [reply web/src/pages/apd/proposed-budget/BudgetSummary.js]: I thought the linter was catching these?

**Review by thetif — APPROVED** [review]

File: web/src/pages/apd/activities/cost-allocation/CostAllocationRows.js [hunk 2]
```diff
@@ -44,32 +59,47 @@ const CostAllocationRows = ({ years, ffy, otherFunding, activityIndex }) => (
...
     {otherFunding && (
       <tr>
         <td className="title" colSpan="4">
           Other Funding Amount
         </td>
         <td>-</td>
-        <td className="budget-table--number">
+        <td
+          className={`budget-table--number ${
+            highlightSubtotals ? 'budget-table--cell__hightlight-lighter' : ''
```
Comment (amyd11) [inline web/src/pages/apd/activities/cost-allocation/CostAllocationRows.js:86]: This is a clever way to handle adding a class. One thing I've used on other projects (and it looks like it's installed in our project already) is this [classnames](https://www.npmjs.com/package/classnames) package. It's a nice way to handle dynamic classes.

You can set up an object for your classes with whatever simple or crazy logic you need for each style class, like:
```
const budgetTableClassName = classNames({
  'budget-table--number': true,
  'budget-table--cell__hightlight-lighter': highlightSubtotals,
  'some-rando-style': shouldShowForA || shouldShowForB
})
```
Then you can just reference your variable as the class and all the logic is prehandled.
`<td className={budgetTableClassName}>`
It's pretty neat!

**Review by amyd11 — APPROVED** [review]
Nice job!

File: web/src/pages/apd/activities/cost-allocation/CostAllocationRows.js (hunk 2 above)
Comment (tbolt) [reply web/src/pages/apd/activities/cost-allocation/CostAllocationRows.js:86]: Ohhh I like that. Will update

**Comment by mirano-darren** [comment]
Looks good to me! Note: Executive summary couldn't be checked (page still crashes) so I checked them in Chromatic

| Given | When | Then | Covered |
| ----- | ---- | ---- | ------- |
|a HITECH or MMIS APD|On the Proposed Budget page, Executive Summary page, Budget and FFP page, and the Activity Schedule Summary page | All changes to tables should only affect the Activity total cost table, Combined Activity Cost table,  State and Contractor Cost breakdown table, Summary Budget Table, Budget and FFP table, Executive Summary Table, Schedule Summary by Activity, and Milestone Tables by Activity | ✅ |
|a HITECH or MMIS APD|On the Proposed Budget page, Executive Summary page, Budget and FFP page, and the Activity Schedule Summary page|If a table has more than 4 rows, each row should alternate in color. Even numbered rows should be gray (#F1F1F1) | ✅ |
|a HITECH or MMIS APD|Tables on the Proposed Budget page and Budget and FFP page| Subtotal rows should have a light yellow background color and Grand total rows should have a dark yellow background color| ✅ |
|a HITECH or MMIS APD|Tables on the Executive Summary page| Total row should be light yellow. This is an exception to the totals/subtotals color scheme. (Verified w/ Akua)| ✅  |
|a HITECH or MMIS APD|On the Proposed Budget page| Help text for the Summary Budget table section should be updated| ✅  |
|a HITECH or MMIS APD| Tables on the Activity Schedule Summary; Combined Activity Costs and State and Contractor Cost Breakdown tables on the Proposed budget | The table header background color should change from black to gray to match the other tables | ✅  |
|a HITECH or MMIS APD| Summary Budget Tables on the Proposed Budget | Federal total should come first before State total | ✅  |
|a HITECH or MMIS APD| Milestone Tables by Activity on the Activity Schedule Summary page | Table header should be changed to the activity name. Subheaders should be Milestone List and Target completion date | ✅ |

**Review by stephanieboydcms — APPROVED** [review]

**Review by jeromeleecms — APPROVED** [review]
Nice - this looks like the Figma. Still other tickets to revise the tables, but this covers the scope of the ticket!

**Review by SGilliamA1M — APPROVED** [review]
Great work on the table updates @tbolt!

**Pull request merged**